Documentation/src/inspect.md

```diff
@@ -0,0 +1,35 @@
+# `IEnumerable` extension method `.Inspect`
+With the `.Inspect` extension method, you can invoke an action for each item in an enumerable, 
+just like `.ForEach` or the `for each` statement would allow you to do, but the method yields the initial enumerable back.
+
+This can be useful when you want to apply a side-effect to a list before returning, or continue selecting on a list after applying a side-effect.
+
+Example 1:

```

Comment:
`Inspect` is primarily intended for debugging or logging, so an example with that in mind would be great

Reply:
Added a comment about logging in 2291d17 . Is that sufficient? @bash